File: app/src/main/java/com/micewine/emu/activities/logAppOutput.java
package com.micewine.emu.activities;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.google.android.material.appbar.CollapsingToolbarLayout;
import com.google.android.material.appbar.MaterialToolbar;
import com.micewine.emu.R;
import com.micewine.emu.core.ShellExecutorCmd;
import com.micewine.emu.databinding.LayoutlogShellOutputBinding;
import com.micewine.emu.viewmodels.ViewModelAppLogs;


public class logAppOutput extends AppCompatActivity {

    private LayoutlogShellOutputBinding binding;
    private ViewModelAppLogs sharedLogs;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = LayoutlogShellOutputBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        ActionBar action = getSupportActionBar();

        if (action != null) {
            action.hide();
        }

        MaterialToolbar toolBar = findViewById(R.id.toolbar_log);

        setSupportActionBar(toolBar);

        CollapsingToolbarLayout collapsingToolBar = findViewById(R.id.toolbar_log_layout);

        collapsingToolBar.setTitle("Logs");

        sharedLogs = new ViewModelProvider(this).get(ViewModelAppLogs.class);

        sharedLogs.setText(ShellExecutorCmd.stdOut);

        sharedLogs.getTextLiveData().observe(this, out -> {
            binding.logShell.setText(out);
        });

    }

    public boolean onCreateOptionsMenu(Menu menu) {


        getMenuInflater().inflate(R.menu.menu_clear, menu);
        return super.onCreateOptionsMenu(menu);
    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == android.R.id.home) {
            finish();
            return true;
        } else if (id == R.id.clear) {

            binding.logShell.setText("");
        }

        return super.onOptionsItemSelected(item);
    }


    @Override
    protected void onDestroy() {
        super.onDestroy();
        this.binding = null;
    }


}

File: app/src/main/java/com/micewine/emu/activities/LogAppOutput.kt
package com.micewine.emu.activities

import android.os.Bundle
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.micewine.emu.R
import com.micewine.emu.core.ShellExecutorCmd.stdErrOut
import com.micewine.emu.databinding.ActivityLogViewerBinding

class LogAppOutput : AppCompatActivity() {
    private var binding: ActivityLogViewerBinding? = null
    private var sharedLogs: ViewModelAppLogs? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLogViewerBinding.inflate(layoutInflater)
        setContentView(binding!!.getRoot())

        val toolBar = findViewById<Toolbar>(R.id.logViewerToolbar)

        toolBar.title = resources.getString(R.string.logTitle)

        val backButton = findViewById<ImageButton>(R.id.backButton)

        backButton?.setOnClickListener {
            finish()
        }

        val clearButton = findViewById<ImageButton>(R.id.clearButton)

        clearButton?.setOnClickListener {
            stdErrOut = ""
            sharedLogs!!.setText("")
        }

        sharedLogs = ViewModelProvider(this)[ViewModelAppLogs::class.java]
        sharedLogs!!.setText(stdErrOut)
        sharedLogs!!.textLiveData.observe(this) { out: String? -> binding!!.logShell.text = out }
    }

    override fun onDestroy() {
        super.onDestroy()
        binding = null
    }

    class ViewModelAppLogs : ViewModel() {
        private val logsText = MutableLiveData<String>()
        val textLiveData: LiveData<String>
            get() = logsText

        fun setText(text: String) {
            logsText.value = text
        }
    }
}
